src/vdaf.rs

```diff
@@ -239,7 +239,10 @@ pub trait Aggregator<const VERIFY_KEY_SIZE: usize, const NONCE_SIZE: usize>: Vda
     ///
     /// Decoding takes a [`Self::PrepareState`] as a parameter; this [`Self::PrepareState`] may be
     /// associated with any aggregator involved in the execution of the VDAF.
-    type PrepareShare: Clone + Debug + ParameterizedDecode<Self::PrepareState> + Encode;
+    type PrepareShare: Clone
+        + Debug
+        + for<'a> ParameterizedDecode<(&'a Self, &'a Self::AggregationParam, usize, u16)>
```

Comment:
Type aliases for the aggregator ID (`usize`) and round number (`u16`) might make this clearer.

Reply:
Type aliases would just help with self-documentation, while newtypes would catch confusion.